### PR TITLE
Balancing with multithreaded recv processes

### PR DESCRIPTION
**Pull request by nritsche**
#355

### CODE REVIEW

**Comment by jrs65** [comment]
Rebuilding now.

File: lib/processes/applyGains.cpp [hunk 1]
```diff
@@ -25,14 +25,16 @@ REGISTER_KOTEKAN_PROCESS(applyGains);
 applyGains::applyGains(Config& config, const string& unique_name,
                        bufferContainer& buffer_container) :
     KotekanProcess(config, unique_name, buffer_container,
-                   std::bind(&applyGains::main_thread, this)) {
+                   std::bind(&applyGains::main_thread, this)),
+    in_buf(get_buffer("in_buf")),
+    out_buf(get_buffer("out_buf")),
+    frame_id_in(frameID(in_buf)),
```
Comment (jrs65) [inline lib/processes/applyGains.cpp]: Does this now work if you just use `frame_id_in(in_buf)` without the intermediate object being constructed?

Comment (nritsche) [reply lib/processes/applyGains.cpp]: Thanks! Fixed.

**Comment by andrerenard** [comment]
Have we decided on if we want to do this as a hot fix, or just merge it in now?  Also it looks like there are now some conflicts with `develop`.

**Comment by andrerenard** [comment]
This needs to be repointed at `master` for the hotfix, and we'll need to fix any strangeness that comes up as a result of that.   Depending on what the tree looks like, it might be easier to start a new branch off master and `patch` these changes into that.

**Comment by nritsche** [comment]
Replaced by #379

**Pull request closed without merging**